isisd/isis_bfd.c

```diff
@@ -255,6 +255,44 @@ static void bfd_debug(int family, union g_addr *dst, union g_addr *src,
 		   command_str, dst_str, interface, src_str);
 }
 
+static void bfd_command(int command, struct bfd_info *bfd_info, int family,
+			void *dst_ip, void *src_ip, char *if_name)
```

Comment:
One required change: please set `args.ifnamelen` otherwise `zclient_bfd_command` will never use it.

```c
		/* Send interface name if any. */
		stream_putc(s, args->ifnamelen);
		if (args->ifnamelen)
			stream_put(s, args->ifname, args->ifnamelen);
```

( https://github.com/FRRouting/frr/blob/master/lib/bfd.c#L481 )

and

Three optional changes here:
- I think we could make `dst_ip`, `src_ip` and `if_name` `const`s.
- Since you are initializing `args` with zero, you don't need to set fields to `0`.
- `args.mhop` is a boolean and even though `0` is `false`, `false` would be clearer

Reply:
Resolved.

ifnamelen is actually set in the lower part of the fun :)